### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by tanveertkd**
This PR adds authentication to the site. You can sign in either by using the test credentials or by creating a new account.  
Test credentials:  
 - email: adarshbalika@gmail.com
 - password: adarshBalika123

### CODE REVIEW

File: src/components/Authentication/Signin.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                        />
+                    </div>
+
+                    <div className="form-input checkbox-reset">
+                        <div>
+                            <input type="checkbox" name="form-checkbox" className="form-checkbox" />
+                            <label for="form-checkbox" className="checkbox-label">
+                                {' '}
+                                Remember me{' '}
+                            </label>
```
Comment (gopalbbharadva) [inline src/components/Authentication/Signin.jsx:69]: Suggestion: U can remove these extra spaces that is generated by prettier

File: src/services/auth-services.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import axios from 'axios';
+
+const signInService = async (email, password) => {
+    const response = await axios.post('/api/auth/login', {
+        email,
+        password,
+    });
+
+    console.log(response);
```
Comment (gopalbbharadva) [inline src/services/auth-services.js]: Can remove unused consoles.

**Review by gopalbbharadva — APPROVED** [review]
Typo: Placeholder for first and last name are for email  in  signup page. just correct that.

P.s: Everything working fine, Nice UI. well structured code.

File: src/components/Authentication/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    return (
+        // <!-- Login body -->
+
+        <div className="login-container">
+            <div className="login">
+                <p className="login-header">Sign Up</p>
+
+                <form
+                    className="login-input-form"
+                    onSubmit={(event) => signUpHandler(event, userInput)}
```
Comment (MiheerTamkhane) [inline src/components/Authentication/Signup.jsx:26]: You can preventDefault here as well no need to pass whole event to a function.

File: src/context/auth-context.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            {children}
+        </AuthContext.Provider>
+    );
+};
+
+const useAuth = () => {
+    const context = useContext(AuthContext);
+    if (context === undefined) throw new Error('Auth context needs to be defined');
+    return context;
+};
```
Comment (MiheerTamkhane) [inline src/context/auth-context.js:54]: Good checks here

File: src/services/auth-services.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    try {
+        if (response.status === 200) return response;
+        else console.log('Failed to login: ', response.status);
```
Comment (MiheerTamkhane) [inline src/services/auth-services.js:11]: I liked how you are also checking for res.status

**Review by MiheerTamkhane — APPROVED** [review]
Code is readable and nicely written. Good to go

File: src/services/auth-services.js (hunk 2 above)
Comment (tanveertkd) [reply src/services/auth-services.js]: Will do. Thanks!

File: src/components/Authentication/Signin.jsx (hunk 1 above)
Comment (tanveertkd) [reply src/components/Authentication/Signin.jsx:69]: Noted. Thanks!

File: src/components/Authentication/Signup.jsx (hunk 3 above)
Comment (tanveertkd) [reply src/components/Authentication/Signup.jsx:26]: Noted. Thank you!

File: src/services/auth-services.js (hunk 5 above)
Comment (tanveertkd) [reply src/services/auth-services.js:11]: Thanks!

**Pull request merged**